consumer/consumer/consumer.go

```diff
@@ -0,0 +1,43 @@
+package consumer
+
+import (
+	"encoding/json"
+	"fmt"
+	"twittertracker/datastore"
+	"twittertracker/parser"
+
+	"github.com/adjust/rmq"
+	"github.com/dghubble/go-twitter/twitter"
+)
+
+// TaskConsumer implements the Consumer interface
+type TaskConsumer struct {
+	DbEnv *datastore.Env
+}
+
+// Consume is the work that will be done by the taskconsumer
+func (consumer *TaskConsumer) Consume(delivery rmq.Delivery) {
+	var tweet twitter.Tweet
+
+	fmt.Println("Printing JSON")
+	payload := delivery.Payload()
+	fmt.Println(payload)
+
+	if err := json.Unmarshal([]byte(payload), &tweet); err != nil {
+		// handle error
+		delivery.Reject()
+		return
+	}
+
+	parsedWords := make(chan string)
+	go parser.GetWordsFromTweet(tweet, parsedWords)
+	saveWord(parsedWords, consumer.DbEnv)
+	delivery.Ack()
+}
+
+func saveWord(parsedWords <-chan string, dbEnv *datastore.Env) {
```

Comment:
can you make the increments one atomic action? If so, we should throw if it ever fails and reject the delivery. If not you can make a call.

Reply:
What do you mean one atomic action? Do you mean increment all words of the tweet at the same time?

Reply:
yeah try to increment all or fail. If it fails that's fine because it just means that it'll be reprocessed. So ideally we want this to be an atomic event.

Reply:
it could be a failure in the machine or the db fails to increment. Any number of factors, but since we're trying the whole tweet as an entry we should treat adding it's data as one operation as well. If the payload fails it automatically gets reprocessed because the payload doesn't get marked as done in the queue. At least that should be how the message queue works. We can do some testing to confirm this.

Reply:
https://redis.io/topics/transactions